main.tf

```
@@ -73,3 +73,75 @@ module "cbr_rule" {
     tags = var.cbr_rules[count.index].tags
   }]
 }
+
+########################################################################################################################
+# SCC Workload Protection Trusted Profile
+########################################################################################################################
+
+# Create Trusted profile for SCC Workload Protection instance
+module "trusted_profile_scc_wp" {
+  count                       = var.cspm_enabled ? 1 : 0
+  source                      = "terraform-ibm-modules/trusted-profile/ibm"
+  version                     = "3.0.0"
+  trusted_profile_name        = var.scc_workload_protection_trusted_profile_name
+  trusted_profile_description = "Trusted Profile for SCC workload Protection instance ${ibm_resource_instance.scc_wp.guid} with required access for configuration aggregator."
+
+  trusted_profile_identity = {
+    identifier    = ibm_resource_instance.scc_wp.crn
+    identity_type = "crn"
+  }
+
+  trusted_profile_policies = [
+    {
+      unique_identifier = "scc-wp"
+      roles             = ["Service Configuration Reader", "Viewer", "Configuration Aggregator Reader"]
+      resources = [{
+        service = "apprapp"
+      }]
+      description = "App Config access"
+    },
+    {
+      unique_identifier = "scc-wp-enterprise"
+      roles             = ["Viewer", "Usage Report Viewer"]
+      resources = [{
+        service = "enterprise"
+      }]
+      description = "Enterprise access"
+    }
+  ]
+
+  trusted_profile_links = [{
+    unique_identifier = "scc-wp-vsi-link"
+    cr_type           = "VSI"
+    links = [{
+      crn = ibm_resource_instance.scc_wp.crn
+    }]
+  }]
+}
+
+################################################################
+# Enable CSPM for SCC Workload Protection instance
+################################################################
+
+# CSPM can only be enabled after the trusted profile exists,
+# but profile can only exist after instance has been created
+# hence we cannot directly enable CSPM in the instance creation
+# and need to use a separate resource to enable it
+resource "restapi_object" "enable_cspm" {
```

maybe change the name of the resource since it will also support disabling cspm
```suggestion
resource "restapi_object" "cspm" {
```